pkg/stores/proxy/proxy_store.go

```diff
@@ -219,6 +214,12 @@ func (s *Store) ByNames(apiOp *types.APIRequest, schema *types.APISchema, names
 	if err != nil {
 		return nil, nil, err
 	}
+	if apiOp.Namespace == "*" {
+		// This happens when you grant namespaced objects with "get" by name in a clusterrolebinding. We will treat
+		// this as an invalid situation instead of listing all objects in the cluster and filtering by name.
+		objs.Items = []unstructured.Unstructured{}
+		return objs, nil, nil
```

Comment:
It's not a great design pattern to return a nil object and a nil error. However I think the calling function should be able to handle `obj.Items` being nil.
```suggestion
		objs = &unstructured.UnstructuredList{}
		return objs, nil, nil
```

Reply:
I purposefully modified the return of a real list call. In this case in particular, I'm trying to preserve a valid `ResourceVersion`, which the [consuming function](https://github.com/rancher/steve/blob/811e0b3572fd1f5008ec3b151658c97d578deca6/pkg/stores/partition/parallel.go#L179) uses.

In short, I'd rather produce a valid (but empty) list call, rather than attempting to construct a minimum 0 value for the unstructuredList that other calls can use.

Reply:
Discussed offline - it seems that it's not necessary to use a real call and other areas (including the UI) can handle that fine. To save on performance, this will be moved to the start of the function and return an empty unstructured list like @KevinJoiner suggested.